### DISCUSSION TITLE
Apply Job has been queued and stuck for multiple days.

### DISCUSSION THREAD

**Discussion by abhinavsquadstack** [terrakube-io/terrakube#627] [Q&A] [answered]
Hey Team,
I have been attempting to execute a 'terraform apply' in one of my workspaces, but the process has been stuck/queue for multiple days. When I try to initiate a new plan, the logs display the following error: `Workspace 0c68da1b-f3ed-4aac-b346-0911fdcec144 is locked. Job creation is denied due to this lock in the <workspace-name>`.

What should I do? From where can I force stop this job? 
I am using terrakube v2.17.0

Thanks

**Comment by alfespa17** [comment, marked as answer]
You can go to the workspace in the UI check the settings and there is an option to lock or unlocked the workspace

**Reply by abhinavsquadstack** [reply]
Thanks for the answer, that unlocked the workspace. But previous job is still in queue state. I guess the 'Cancel Job' button doesn't appear when a job is in the queue state. <img width="816" alt="image" src="https://github.com/AzBuilder/terrakube/assets/121935932/9a96456d-1318-438d-893a-d49e916c8b86">

**Reply by alfespa17** [reply]
You will have to connect to the database and update the job table and the status to cancelled, I still need to add some logic to cancel jobs that have been running for more than 6 hours for some reaseon.

**Comment by jcanizalez** [comment]
> You can go to the workspace in the UI check the settings and there is an option to lock or unlocked the workspace

I have created #628 to improve the visibility for the unlock/lock in the UI
